src/catch.rs

```diff
@@ -174,12 +176,33 @@ fn match_npm(line: &str) -> Result<Vec<Package>, Box<dyn Error>> {
     )
 }
 
+//cache for args
+lazy_static! {
```

Comment:
Why did you wrap make this static? I think it degrades the performance since now everything is inside a mutex and it constructs a hashmap.

Reply:
I've also been meaning to rewrite the regex part with a library like nom, which has a lot better performance.

Reply:
> Why did you wrap make this static? I think it degrades the performance since now everything is inside a mutex and it constructs a hashmap.

I have measured this part with flamegraph, and previously 87% was spent on `Regex::new()`. Now just 6% of time is spent on all parsing, and other time is spent on operations with git.
Hashmap is constructed on first call to it, thanks to lazy_static. Mutex is used to constraint requirements for static variables it couldn't block, cause code is single threaded. 
> I've also been meaning to rewrite the regex part with a library like nom, which has a lot better performance.
It has, but you're spending very little time on parsing.

Reply:
Moreover, look at  [this](https://docs.rs/regex/1.3.7/regex/#example-avoid-compiling-the-same-regex-in-a-loop). My solution is more transparent, because it caches every call in hashmap

Reply:
What part did you test with flamegraph? Because the only codepath that really matters in regards to performance in `emplace catch`, this is called every time you enter an install command on the command line and when it's slow it will influence the speed of your terminal.

Not using `lazy_static` was a conscious decision because it adds overhead to this.

Reply:
It doesn't matter that much in this case because I'll change the regex system to use nom anyway.

Reply:
i'll make a criterion bech for it :) On flamegraph it take 15% of time for regex build. Allocations are invisible.

Reply:
It's need conversion to separate lib.rs and i'm lazy doing it. But on my mind, there are 2 memory allocations and hash calculations for str, which is very cheap